compiler/rustc_hir_analysis/src/collect/lifetimes.rs

```diff
@@ -24,42 +24,44 @@ use rustc_span::Span;
 use std::fmt;
```

Comment:
Perhaps @cjgillot has opinions of whether the late lifetime resolution machinery here should also be used to resolve `for<T>` binders?

Reply:
And if so, what the new naming scheme for all these items should be :skull:

Reply:
The logic in this file should pretty much work out of the box on type & const binders. There is a question about naming, I agree.
About naming: "late resolution" denotes what `rustc_resolve::late` does. This file is about computing debruijn indices and in-binder index.

Reply:
This file also handles other region kinds, right? Otherwise I guess I could rename it to like.. `collect/resolve_bound_vars.rs` or something?